phable_cli/cli.py

```diff
@@ -363,6 +342,51 @@ def subscribe_to_task(ctx, task_ids: list[int]):
         client.add_user_to_task_subscribers(task_id=task_id, user_phid=user["phid"])
 
 
+@cli.command(name="report-done-tasks")
+@click.option(
+    "--milestone/--no-milestone",
+    default=False,
+    help=(
+            "If --milestone is passed, the task will be moved onto the current project's associated "
+            "milestone board, instead of the project board itself"
+    ),
+)
+@click.option(
+    "--format",
+    type=click.Choice(("plain", "json")),
+    default="plain",
+    help="Output format",
+)
+@click.option(
+    "--source",
+    type="str",
+    default="Done",
+    help="",
+)
+@click.option(
+    "--destination",
+    type="str",
+    default="Reported",
+    help="",
+)
+def report_done_tasks(milestone: bool, format:str, source: str, destination: str):
+    """Print the details of all tasks in the `from` column and move them to the `to` column.
+
+    This is used to produce the weekly reports, and document the tasks as reported once the report is done."""
+    client = PhabricatorClient()
+    target_project_phid = client.get_main_project_or_milestone(
+        milestone, config.phabricator_default_project_phid
+    )
+    column_source_phid = client.find_column_in_project(target_project_phid, source)
+    column_destination_phid = client.find_column_in_project(target_project_phid, destination)
+    tasks = client.find_tasks_in_column(column_source_phid)
+    for task in tasks:
+        click.echo("="*50)
+        client.enrich_task(task)
+        echo_task(click.echo, format, task)
+        client.move_task_to_column(task['id'], column_destination_phid)
```

Comment:
The issue here is that this breaks JSON formatting. I would probably do something like:
```suggestion
    tasks = client.find_tasks_in_column(column_source_phid)
    tasks = [client.enrich_task(task) for task in tasks]
    for task in task:
        client.move_task_to_column(task["id"], column_destination_phid)
    if format == "plain":
        for task in tasks:
            echo_task(click.echo, format, task)
            click.echo("=" * 50)
    else:
        click.echo(json.dumps(tasks, indent=2))
```

Reply:
Slightly different proposal in the next push, so as to not materialize the full list of enriched tasks, and to move all of the display logic to the same place. It would probably make sense to introduce a concept of task display and to allow to plug multiple different implementation (out of scope for this MR).